be/src/storage/rowset/vectorized/segment_iterator.cpp

```diff
@@ -820,40 +820,43 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
         chunk_start = next_start;
         DCHECK_EQ(chunk_start, chunk->num_rows());
     }
-    chunk_start = _filter_by_expr_predicates(chunk, rowid);
+
+    size_t raw_chunk_size = chunk->num_rows();
+
+    size_t chunk_size = _filter_by_expr_predicates(chunk, rowid);
 
     _opts.stats->block_load_ns += sw.elapsed_time();
 
     int64_t total_read = _opts.stats->raw_rows_read - prev_raw_read;
 
-    DCHECK_EQ(chunk_start, chunk->num_rows());
-    if (UNLIKELY(chunk_start == 0)) {
+    if (UNLIKELY(raw_chunk_size == 0)) {
```

Comment:
I think we could move this check to line 823, then we don't need the raw_chunk_size variate?

Reply:
It's okay. But then `block_load_ns` will not be updated.  I was thinking it's better to make metrics as accurate as possible.

```
    _opts.stats->block_load_ns += sw.elapsed_time();
```